### DISCUSSION TITLE
Projections consisting of a single ray (listmode)

### DISCUSSION THREAD

**Discussion by BOle-ProZ** [CERN/TIGRE#431] [Q&A]
Hello TIGRE community,

I have a question regarding the SIRT algorithm (or any other to solve my problem).
Usually, when talking about a projection from a certain angle, a 2D "look through" projection is meant. Is it also possible to reconstruct a volume from an arrangement of rays with known trajectories through the sample? Each ray is independant and has a brightness value associated to it. I added an image to illustrate this problem.

Please don't hesitate to ask for clarification if needed.

Thank You and have a nice Day!
Ole
![3d with random rays](https://user-images.githubusercontent.com/94117204/227480918-1d99edce-76b6-444c-9cab-a2265be7a8bd.PNG)

**Comment by AnderBiguri** [comment]
Hi @BOle-ProZ ,


I see, you are asking if its possible to do listmode reconstruction, essentially, right? Is this for PET, possibly? 

The answer is no, unfortunately. Listmode requires a different setup for the projectors, and a different geometry description (2 points per line of response etc). Unfortunately TIGRE does not have that implemented.  Have a look at STIR, it certainly has that (and more PET features). 

However, if you or anyone around you is willing to put the time, I would be happy to guide you on what exactly is needed to change in the operators to make the listmode compatible. This is quite a big tasks, of course.

**Reply by BOle-ProZ** [reply]
Hello @AnderBiguri ,
 
Thank you for the quick reply! It is not a typical CT problem. We use an optical setup with a laser to measure fluorescence and absorption in very small samples. 

What is STIR? A different tool box for tomography?

**Reply by AnderBiguri** [reply]
Ah I see.  STIR is a different tool, yes, but it is explicitly for PET reconstruction, so while it can forward project listmode data, I am not sure you can use with data that is not from PET. I mean, you might, but I suspect it will be quite hard, its not an easy tool to use at all.
